### PR TITLE
Hotfix/e UI 4873 SSCS - Dynamic lists in a collection within a complex type

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-4873

### Change description ###

Reapplied dynamic list values for mid event callbacks using page wizard

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/services/fields/fields.utils.ts [hunk 1]
```diff
@@ -242,6 +247,82 @@ export class FieldsUtils {
...
+   * down the tree. Server response returns data only in
+   * the `value` object of parent complex type
+   *
+   * EUI-2530 Dynamic Lists for Elements in a Complex Type
+   *
+   * @param jsonBody - { case_fields: [ CaseField, CaseField ] }
+   */
+   public static handleNestedDynamicLists(jsonBody: { case_fields: CaseField[] }): any {
+
+    if (jsonBody.case_fields) {
```
Comment (paulhowes-HMCTS) [inline src/shared/services/fields/fields.utils.ts:262]: Is the if statement necessary? i.e. if there are none it just won't go into the foreach.

**Review by paulhowes-HMCTS — APPROVED** [review]
Just a minor tweak if you feel compelled to do it but no show stoppers by the look of it.

**Review by Mo-Lala84 — APPROVED** [review]
approved

File: src/shared/services/fields/fields.utils.ts (hunk 1 above)
Comment (adnanakgun) [reply src/shared/services/fields/fields.utils.ts:262]: I copy-pasted this code from CasesService to a more common location. Additionally, I sort of remember `foreach` causing an error when `case_fields` doesn't exist.

**Pull request merged**